File: WebAPI/Models/Entities/Brand.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace BlockRouter.WebAPI.Models.Entities;

[Serializable]
public partial class Brand
{
    [Required]
    public int Id { get; set; }

    [Required]
    [StringLength(32, MinimumLength = 2)]
    public string Name { get; set; } = null!;

    [Required]
    public bool IsActive { get; set; }

    [JsonIgnore]
    public virtual ICollection<Model> Models { get; } = new List<Model>();
}

File: Client/src/models/entities/Brand.ts
/**
 * Reproduction of entity 'Brand'.
 */
export default class Brand {
  public id: number;

  public name: string;

  public isActive: boolean;

  public constructor(id: number, name: string, isActive: boolean) {
    this.id = id;
    this.name = name;
    this.isActive = isActive;
  }
}
